src/i2c/dma.rs

```diff
@@ -601,6 +603,36 @@ where
     }
 }
 
+impl<I2C, TX_TRANSFER, RX_TRANSFER> ErrorType for I2CMasterDma<I2C, TX_TRANSFER, RX_TRANSFER>
+where
+    I2C: Instance,
+    TX_TRANSFER: DMATransfer<&'static [u8]>,
+    RX_TRANSFER: DMATransfer<&'static mut [u8]>,
+{
+    type Error = super::Error;
+}
+
+impl<I2C, TX_TRANSFER, RX_TRANSFER> i2c::I2c for I2CMasterDma<I2C, TX_TRANSFER, RX_TRANSFER>
+where
+    I2C: Instance,
+    TX_TRANSFER: DMATransfer<&'static [u8]>,
+    RX_TRANSFER: DMATransfer<&'static mut [u8]>,
+{
+    fn transaction(
+        &mut self,
+        addr: u8,
+        operations: &mut [Operation<'_>],
+    ) -> Result<(), Self::Error> {
+        for operation in operations {
+            match operation {
+                Operation::Read(dest) => self.hal_i2c.read(addr, dest),
```

Comment:
This annoying me. I've newer used `I2C+DMA`. Is this correct use call inner implementation? Does it actually use DMA?

Reply:
Could it be explained more widely in docs?

> same as the [methods here](https://github.com/stm32-rs/stm32f4xx-hal/blob/master/src/i2c/dma.rs#L540-L581).

Yeah. These methods disturb me too.